### PR TITLE
feat: sync team_components table with the deletion of components & teams

### PR DESCRIPTION
**Pull request by adityathebe**
Resolves: https://github.com/flanksource/incident-commander/issues/408

### CODE REVIEW

File: views/007_events.sql [hunk 1]
```diff
@@ -195,4 +199,21 @@ $$ LANGUAGE plpgsql;
...
\ No newline at end of file
+EXECUTE PROCEDURE notifications_trigger_function ();
+
+-- Handle component updates
+CREATE
+OR REPLACE FUNCTION handle_component_updates () RETURNS TRIGGER AS $$
+BEGIN
+  IF OLD.deleted_at IS NULL AND NEW.deleted_at IS NOT NULL THEN
+    DELETE FROM team_components WHERE component_id = OLD.id;
+  END IF;
```
Comment (yashmehrotra) [inline views/007_events.sql:210]: Can we also do deletions for component relationships and other related resources as well ?

Comment (yashmehrotra) [reply views/007_events.sql:210]: Also, I think these should be in a different file ? Like how we created a new one for canaries/check deletion handling

Doesn't belong in events.sql

**Review by yashmehrotra — APPROVED** [review]
:shipit:

File: views/018_component_triggers.sql [hunk 2]
```diff
@@ -0,0 +1,22 @@
+CREATE
+OR REPLACE FUNCTION handle_component_updates () RETURNS TRIGGER AS $$
+BEGIN
+  IF OLD.deleted_at IS NULL AND NEW.deleted_at IS NOT NULL THEN
+    DELETE FROM team_components WHERE component_id = OLD.id;
```
Comment (moshloop) [inline views/018_component_triggers.sql:5]: I am not sure we should be doing this - If I am looking at a deleted component, I still want to see what it was related to before it was deleted. 

A relationship should only be deleted when it is no longer true e.g. An EBS volume is detached from an ec2instance - in which case the component/config relationship.deleted_at should reflect the detach time

File: views/019_teams_triggers.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
+CREATE
+OR REPLACE FUNCTION handle_team_updates () RETURNS TRIGGER AS $$
+BEGIN
+  IF OLD.deleted_at IS NULL AND NEW.deleted_at IS NOT NULL THEN
+    DELETE FROM team_components WHERE team_id = OLD.id;
```
Comment (moshloop) [inline views/019_teams_triggers.sql:5]: This is ok, when deleting a team they no longer "own" any components.

File: views/018_component_triggers.sql (hunk 2 above)
Comment (moshloop) [reply views/018_component_triggers.sql:5]: @adityathebe  / @yashmehrotra  ^

**Review by moshloop — CHANGES_REQUESTED** [review]

**Review by moshloop — APPROVED** [review]

**Pull request merged**